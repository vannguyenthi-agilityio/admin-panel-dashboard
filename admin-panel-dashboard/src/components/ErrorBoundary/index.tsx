import {
  Component,
  ErrorInfo,
  ReactNode
} from 'react';
import { Link } from 'react-router-dom';

interface Props {
  children: ReactNode
}

interface State {
  hasError: boolean
}

class ErrorBoundary extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(): State  {
    // Update state so the next render will show the fallback UI.
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo): void {
    console.error('Uncaught error:', error, errorInfo);
  }

  render(): ReactNode {
     if (this.state.hasError) {
      return (
        <div className='flex items-center justify-center min-h-screen bg-gray-100'>
        <div className='p-10 bg-white rounded shadow-lg text-center'>
          <h1 className='text-4xl font-bold mb-4 text-red-600'>Oops!</h1>
          <p className='text-xl mb-4'>Something went wrong.</p>
          <p className='text-gray-600 mb-8'>We're sorry for the inconvenience. Please try refreshing the page or click the button below to go back to the homepage.</p>
          <Link to='/' className='px-6 py-2 bg-blue-600 text-white rounded hover:bg-blue-700'>
            Go to Homepage
          </Link>
        </div>
      </div>
      )
     }

     return this.props.children;
  }
}

export default ErrorBoundary
