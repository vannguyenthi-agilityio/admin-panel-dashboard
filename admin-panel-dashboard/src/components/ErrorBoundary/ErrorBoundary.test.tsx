import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import ErrorBoundary from '.';

describe('Error Boundary Component', () => {
  test('renders error boundary component', () => {
    const {container} = render(
      <MemoryRouter>
        <ErrorBoundary>
          <div>Content without error</div>
        </ErrorBoundary>
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  })
})
