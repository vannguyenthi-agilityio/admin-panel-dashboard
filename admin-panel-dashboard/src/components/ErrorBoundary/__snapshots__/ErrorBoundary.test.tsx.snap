// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Boundary Component renders error boundary component 1`] = `
<div>
  <div>
    Content without error
  </div>
</div>
`;
